physical/etcd2.go

```diff
@@ -0,0 +1,601 @@
+package physical
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	metrics "github.com/armon/go-metrics"
+	"github.com/coreos/etcd/client"
+	"github.com/coreos/etcd/pkg/transport"
+	log "github.com/mgutz/logxi/v1"
+)
+
+const (
+	// Ideally, this prefix would match the "_" used in the file backend, but
+	// that prefix has special meaining in etcd. Specifically, it excludes those
+	// entries from directory listings.
+	EtcdNodeFilePrefix = "."
+
+	// The lock prefix can (and probably should) cause an entry to be excluded
+	// from diretory listings, so "_" works here.
+	EtcdNodeLockPrefix = "_"
+
+	// The delimiter is the same as the `-C` flag of etcdctl.
+	EtcdMachineDelimiter = ","
+
+	// The lock TTL matches the default that Consul API uses, 15 seconds.
+	EtcdLockTTL = 15 * time.Second
+
+	// The amount of time to wait between the semaphore key renewals
+	EtcdLockRenewInterval = 5 * time.Second
+
+	// The amount of time to wait if a watch fails before trying again.
+	EtcdWatchRetryInterval = time.Second
+
+	// The number of times to re-try a failed watch before signaling that leadership is lost.
+	EtcdWatchRetryMax = 5
+)
+
+// EtcdBackend is a physical backend that stores data at specific
+// prefix within Etcd. It is used for most production situations as
+// it allows Vault to run on multiple machines in a highly-available manner.
+type EtcdBackend struct {
```

Comment:
Given that the "official" etcd backend going forward will be the etcd3 one that you are maintaining, I think this should be called `Etcd2Backend` and the current `Etcd3Backend` should remain as-is or be the `EtcdBackend`. This just makes it clearer to viewers of the code that either they're side-by-side or that the 3 one is the "official" one. Minor optics, but those are nice!  :-)